src/ElasticSearchStore.ts

```diff
@@ -0,0 +1,284 @@
+          reply(err)
+        }
+      }
+    },
+
+    remove: async function (this: any, msg: any, reply: any) {
+      const ent = msg.ent
+      const idToBeRemoved = msg.q.id
+      const index = resolveIndex(ent, options)
+
+      try {
+        await client.delete({
+          index,
+          id: idToBeRemoved,
+          refresh: 'wait_for',
+        })
+        reply(null, ent)
+      } catch (err) {
+        console.error('Error in removing document:', err)
+        reply(err)
+      }
+    },
+
+    close: function (this: any, _msg: any, reply: any) {
+      this.log.debug('close', desc)
+      reply()
+    },
+
+    native: function (this: any, _msg: any, reply: any) {
+      reply(null, {
+        client: () => client,
+      })
+    },
+  }
+
+  init(seneca, options, store)
+  // desc = meta.desc
+
+  seneca.prepare(async function () {
+    try {
+      client = new Client({
+        node: options.elasticsearch.node,
+        auth: options.auth,
+      })
+    } catch (error) {
+      console.error('Failed to initialize ElasticSearch client:', error)
+    }
+  })
+
+  return {
+    name: store.name,
+    // tag: meta.tag,
+    exportmap: {
+      native: () => {
+        return { client }
+      },
+    },
+  }
+}
+
+function resolveIndex(ent: any, options: ElasticSearchStoreOptions): string {
+  let indexOpts = options.index
+  if ('' != indexOpts.exact && null != indexOpts.exact) {
+    return indexOpts.exact
+  }
+
+  let canonstr = ent.canon$({ string: true })
+  indexOpts.map = indexOpts.map || {}
+  if ('' != indexOpts.map[canonstr] && null != indexOpts.map[canonstr]) {
+    return indexOpts.map[canonstr]
+  }
+
+  let prefix = indexOpts.prefix
+  let suffix = indexOpts.suffix
+
+  prefix = '' == prefix || null == prefix ? '' : prefix + '_'
+  suffix = '' == suffix || null == suffix ? '' : '_' + suffix
+
+  let infix = ent
+    .canon$({ string: true })
+    .replace(/-\//g, '')
+    .replace(/\//g, '_')
+
+  return prefix + infix + suffix
+}
+
+function buildQuery(cleanedQuery: any) {
+  const boolQuery: any = { must: [], filter: [] }
+  Object.keys(cleanedQuery).forEach((key) => {
+    if ('vector' !== key) {
+      boolQuery.filter.push({ term: { [key]: cleanedQuery[key] } })
+    }
+  })
+  return { bool: boolQuery }
+}
+
+function isKnnSearch(query: any) {
+  return !!(query.directive$ && query.directive$.vector$ && query.vector)
+}
+
+async function executeKnnSearch(client: any, index: any, q: any, query: any) {
+  const knnResponse = await client.knnSearch({
+    index,
+    knn: {
+      field: 'vector',
+      query_vector: q.vector,
+      k: q.directive$.vector$.k || 10,
+      num_candidates: 100,
+    },
+    filter: query.bool.filter.length ? query : undefined,
+  })
+
+  const { hits } = knnResponse
+  return hits.hits
+    .filter((hit: any) =>  0.5 <= hit._score) // Adjust the threshold based on your similarity measure
+    .map((hit: any) => ({ id: hit._id, ...hit._source }))
```

Comment:
Remove the `Array.prototype.filter` and in the `Array.prototype.map`, you can both entize and set the custom$.score.

`{ ..., custom$: { score: hit._score } }`

Also, make sure you unit test this.

Reply:
Done, please check about these changes to see if its ok.